web-server/src/comment/comment.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+    Body,
+    Controller,
+    Get,
+    Query,
+    Post as HttpPost,
+    ValidationPipe,
+    HttpException,
+    HttpStatus
+} from '@nestjs/common';
+import { Repository, TreeRepository } from 'typeorm';
+import { Comment } from './comment.entity';
+import { CommentDto } from './comment.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Post } from '../post/post.entity';
+
+@Controller('comments')
+export class CommentController {
+    constructor(
+        @InjectRepository(Comment) private readonly commentRepository: TreeRepository<Comment>,
+        @InjectRepository(Post) private readonly postRepository: Repository<Post>
+
+    ) { }
+
+    @HttpPost()
+    async create(
+        @Body(
+            new ValidationPipe({
+                whitelist: true,
+                transform: true,
+            }),
+        )
+        data: CommentDto,
+    ): Promise<Comment> {
+        const post: Post = await this.postRepository.findOne(data.postId);
+
+        if (post === undefined) {
+            throw new HttpException('Incorrect postId submitted.', HttpStatus.BAD_REQUEST);
```

Comment:
also how can we unit test this?

Reply:
With the current approach, I'd have to mock the repo call.
In reality, The repo call would be in a service and I would unit test that functionality there. 
In true YAGNI fashion, I decided against adding that much boilerplate code for just 4 lines of code, so no service. 

I'll add a simple unit test.

Reply:
sounds good.

For future reference, exactly because of this conundrum I ended up spinning up a mini database per testcase and doing api testing for happy paths.

here is gist straight out of our mono repo

https://gist.github.com/tetrohed/5e9a3c122007791e588c211448453ec1

for the `services/edge cases/fail paths` I still do unit tests.